app/front-end/src/features/editor/components/toolbarView/toolbarGroupButtons/mergeGroupButtons.tsx

```diff
@@ -1,26 +1,117 @@
+import { ToolbarGroupItem, ToolbarGroupItemProps } from '@/features/editor/components/toolbarView';
+import { useToolbarContext } from '@/features/editor/hooks';
+import { useStatusContext } from '@/hooks';
 import { MergeType as MergeTypeIcon } from '@mui/icons-material';
+import { useCallback, useMemo } from 'react';
 
-import { ToolbarGroupItemProps } from '@/features/editor/components/toolbarView';
+export interface MergeGroupButtonsProps {}
 
-const mergeLovdAndGnomadClick = () => {
-  console.log('Clicked Merge LOVD & gnomAD Button!');
-};
+export const MergeGroupButtons: React.FC<MergeGroupButtonsProps> = () => {
+  const { blockedStateUpdate } = useStatusContext();
+  const {
+    saveTo,
+    override,
+    lovdFile,
+    clinvarFile,
+    gnomadFile,
+    lovdErrorStateUpdate,
+    clinvarErrorStateUpdate,
+    gnomadErrorStateUpdate,
+  } = useToolbarContext();
 
-const mergeLovdAndClinvarClick = () => {
-  console.log('Clicked Merge LOVD & ClinVar Button!');
-};
+  const mergeLovdAndGnomadClick = useCallback(async () => {
+    clinvarErrorStateUpdate('');
+
+    if (!lovdFile) {
+      lovdErrorStateUpdate('Please select a LOVD file');
+      return;
+    }
+
+    if (!gnomadFile) {
+      gnomadErrorStateUpdate('Please select a gnomAD file');
+      return;
+    }
+
+    blockedStateUpdate(true);
+
+    try {
+      console.log(
+        'Clicked Merge LOVD & gnomAD Button! Params:\n  saveTo:',
+        saveTo,
+        '\n  override:',
+        override,
+        '\n  lovd:',
+        lovdFile,
+        '\n  gnomad:',
+        gnomadFile
+      );
+
+      await new Promise((resolve) => setTimeout(resolve, 1000)); // TODO: remove this line
+    } catch (error) {
+      console.error('Error merging LOVD & gnomAD files:', error);
+    } finally {
+      blockedStateUpdate(false);
+    }
+  }, [saveTo, override, lovdFile, gnomadFile]);
+
+  const mergeLovdAndClinvarClick = useCallback(async () => {
+    gnomadErrorStateUpdate('');
 
-export const MergeGroupButtons: ToolbarGroupItemProps[] = [
-  {
-    group: 'merge',
-    icon: MergeTypeIcon,
-    label: 'Merge LOVD & gnomAD',
-    onClick: mergeLovdAndGnomadClick,
-  },
-  {
-    group: 'merge',
-    icon: MergeTypeIcon,
-    label: 'Merge LOVD & ClinVar',
-    onClick: mergeLovdAndClinvarClick,
-  },
-];
+    if (!lovdFile) {
+      lovdErrorStateUpdate('Please select a LOVD file');
+      return;
+    }
+
+    if (!clinvarFile) {
+      clinvarErrorStateUpdate('Please select a ClinVar file');
+      return;
+    }
```

Comment:
If no files are selected and a button is clicked, it only highlights the first file as missing. Would be better to remove both return checks and after that add if (!lovdFile|| !clinvarFile) return;


In addition, the 'Please select a X file' should be removed since it's not fully readable anyways. Keeping the selection component the same and just changing the border color and text color to red should be enough.

Reply:
Fixed with [MDE/PKFE-31 select changes during file selection](https://github.com/Strexas/kath/pull/57/commits/f758a51ceb9e6c66867e58cfa3697ddbaab254a8)